steps/test_expval.py

```diff
@@ -0,0 +1,229 @@
+        contains a Hadamard gate."""
+        lst = []
+
+        hadamard_qasm = 'OPENQASM 2.0;\ninclude "qelib1.inc";\nqreg q[2];\ncreg c[2];\nh q[0];\n'
+        op = '["[Z0]"]'
+
+        # Instead of creating a json at the end of the step, just append it to
+        # a local list
+        monkeypatch.setattr(expval, "save_list", lambda val, name: lst.append(val))
+
+        expval.run_circuit_and_get_expval(backend_specs, hadamard_qasm, op)
+        assert math.isclose(lst[0][0], 0.0, abs_tol=tol)
+
+    def test_run_circuit_and_get_expval_rot_cnot(self, backend_specs, monkeypatch):
+        """Tests that the correct result in obtained for a circuit that
+        contains rotations and a CNOT."""
+        lst = []
+
+        theta = 0.432
+        phi = 0.123
+
+        circ_qasm = f'OPENQASM 2.0;\ninclude "qelib1.inc";\nqreg q[2];\ncreg c[2];\nrx({theta}) q[0];\nrx({phi}) q[1];\ncx q[0],q[1];'
+
+        op = '["[Z0]", "[Z1]"]'
+
+        # Instead of creating a json at the end of the step, just append it to
+        # a local list
+        monkeypatch.setattr(expval, "save_list", lambda val, name: lst.append(val))
+
+        expval.run_circuit_and_get_expval(backend_specs, circ_qasm, op)
+
+        res = np.array(lst[0])
+        assert np.allclose(res, np.array([np.cos(theta), np.cos(theta) * np.cos(phi)]), atol=tol)
+
+    @pytest.mark.xfail
+    def test_hadamard_expectation_plugin_integration(self, backend_specs, monkeypatch):
+        """Test that the expectation value of the Hadamard is computed
+        correctly when integrating the step function with the plugin."""
+        n_wires = 2
+
+        theta = 0.432
+        phi = 0.123
+
+        dev = qml.device("orquestra.qulacs", wires=2)
+
+        def circuit():
+            qml.RY(theta, wires=[0])
+            qml.RY(phi, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.Hadamard(wires=0)), qml.expval(qml.Hadamard(wires=1))
+
+        qnode = qml.QNode(circuit, dev)
+        qnode._construct([], {})
+        qasm_circuit = dev.serialize_circuit(qnode.circuit)
+        ops, _ = dev.process_observables(qnode.circuit.observables)
+        ops = json.dumps(ops)
+
+        lst = []
+
+        # Instead of creating a json at the end of the step, just append it to
+        # a local list
+        monkeypatch.setattr(expval, "save_list", lambda val, name: lst.append(val))
+        expval.run_circuit_and_get_expval(dev.backend_specs, qasm_circuit, ops)
+        expected = np.array(
+            [np.sin(theta) * np.sin(phi) + np.cos(theta), np.cos(theta) * np.cos(phi) + np.sin(phi)]
+        ) / np.sqrt(2)
+
+        assert np.allclose(lst[0], expected, atol=analytic_tol)
+
+
+@pytest.fixture
+def token():
+    """Get the IBMQX test token."""
+    t = os.getenv("IBMQX_TOKEN_TEST", None)
+
+    if t is None:
+        pytest.skip("Skipping test, no IBMQ token available")
+
+    yield t
+    IBMQ.disable_account()
```

Comment:
Nice! :fire:

Reply:
Does this mean that this test is skipped in the CI (or is there somehow a token there)?

Reply:
That's a good catch! It's skipped for now but will look into configuring it for the CI.